support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -75,24 +78,36 @@ class ZuoraSubscriptionCreator(
 
     val environment = TouchPointEnvironments.fromStage(Configuration.stage, state.user.isTestUser)
 
+    val maybeGiftRecipWithCode = state.giftRecipient.map(maybeAddCode(state.product.billingPeriod, _))
```

Comment:
```suggestion
    val maybeGiftRecipientWithCode = state.giftRecipient.map(maybeAddCode(state.product.billingPeriod, _))
```

Reply:
I need to avoid the habit of shortening things.  I will have a look through the whole PR for these kind of things.

Reply:
`maybeGiftRecipientWithCode` is a bit low level anyway, the type is already `GiftRecipient` in reflection of that.
Also I will rename `maybeAddCode` to `addGiftPurchaseDetails` as follows
```scala
    val maybeGiftPurchase = state.giftRecipient.map(addGiftPurchaseDetails(state.product.billingPeriod, _))
```